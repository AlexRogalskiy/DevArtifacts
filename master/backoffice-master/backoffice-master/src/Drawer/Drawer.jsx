import React from 'react'
import PropTypes from 'prop-types'

import {
  Drawer as MaterialDrawer,
  Divider,
  IconButton,
  withStyles,
} from '@material-ui/core'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'

import Menu from '../Menu'

const drawerWidth = 280

const styles = theme => ({
  drawerPaper: {
    position: 'relative',
    height: '100%',
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    height: theme.spacing.unit * 8,
  },
})

const Drawer = ({
  data,
  isOpen,
  onClose,
  redirectTo,
  classes,
}) => (
  <MaterialDrawer
    variant="persistent"
    classes={{
      paper: classes.drawerPaper,
    }}
    anchor="left"
    open={isOpen}
  >
    <div className={classes.drawerInner}>
      <div className={classes.drawerHeader}>
        <IconButton onClick={onClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>

      <Divider />

      <Menu data={data} redirectTo={redirectTo} />
    </div>
  </MaterialDrawer>
)


Drawer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  isOpen: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  redirectTo: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

Drawer.defaultProps = {
  isOpen: false,
}

export default withStyles(styles)(Drawer)
