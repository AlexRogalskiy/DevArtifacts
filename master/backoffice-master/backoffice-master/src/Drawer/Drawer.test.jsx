import React from 'react'
import { mount } from 'enzyme'

import menuData from '../../__tests__/data/menu'

import Drawer from './'

it('renders correctly', () => {
  const tree = mount(<Drawer
    isOpen
    data={menuData}
    onClose={() => {}}
    redirectTo={() => {}}
  />)

  expect(tree).toMatchSnapshot()
})
