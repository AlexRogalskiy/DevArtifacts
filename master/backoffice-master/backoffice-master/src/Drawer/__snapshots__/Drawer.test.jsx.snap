// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiDrawer-docked-3"
>
  <div
    className="MuiPaper-root-14 MuiPaper-elevation0-16 MuiDrawer-paper-4 Drawer-drawerPaper-1 MuiDrawer-paperAnchorLeft-5 MuiDrawer-paperAnchorDockedLeft-9"
    direction="right"
    style={Object {}}
  >
    <div>
      <div
        className="Drawer-drawerHeader-2"
      >
        <button
          className="MuiButtonBase-root-47 MuiIconButton-root-41"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-46"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-50"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-57"
          />
        </button>
      </div>
      <hr
        className="MuiDivider-root-64"
      />
      <div
        className="Menu-root-68"
      >
        <ul
          className="MuiList-root-69 MuiList-padding-70"
        >
          <div
            className="MuiButtonBase-root-47 MuiListItem-root-74 MuiListItem-default-77 MuiListItem-gutters-81 MuiListItem-button-82"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemText-root-85"
            >
              <span
                className="MuiTypography-root-91 MuiTypography-subheading-98 MuiListItemText-primary-88"
              >
                Dashboard
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-57"
            />
          </div>
          <div
            className="MuiButtonBase-root-47 MuiListItem-root-74 MuiListItem-default-77 MuiListItem-gutters-81 MuiListItem-button-82 MenuItem-isDisabled-73"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemText-root-85"
            >
              <span
                className="MuiTypography-root-91 MuiTypography-subheading-98 MuiListItemText-primary-88"
              >
                New portfolio
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-57"
            />
          </div>
          <hr
            className="MuiDivider-root-64"
          />
          <div
            className="MuiButtonBase-root-47 MuiListItem-root-74 MuiListItem-default-77 MuiListItem-gutters-81 MuiListItem-button-82"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-50 MuiListItemIcon-root-116"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
              />
            </svg>
            <div
              className="MuiListItemText-root-85"
            >
              <span
                className="MuiTypography-root-91 MuiTypography-subheading-98 MuiListItemText-primary-88"
              >
                Report a bug
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-57"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
